dist/stat_data.py

```diff
@@ -335,6 +335,46 @@ def __init__(self, name, desc, flags=''):
     CapacityStat('fsync_all_time', 'background fsync time (msecs)', 'no_clear,no_scale'),
 
+    ##########################################
+    # Checkpoint statistics
+    ##########################################
+    CheckpointStat('checkpoints', 'number of checkpoints started'),
+    CheckpointStat('checkpoint_custom_dsrc', 'number of custom data sources checkpointed'),
+    CheckpointStat('checkpoint_fsync_post', 'transaction fsync calls for checkpoint after allocating the transaction ID'),
+    CheckpointStat('checkpoint_fsync_post_duration', 'transaction fsync duration for checkpoint after allocating the transaction ID (usecs)', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_generation', 'transaction checkpoint generation', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_handle_applied', 'transaction checkpoint most recent handles applied'),
+    CheckpointStat('checkpoint_handle_checked', 'transaction checkpoint data handle operation applied'),
+    CheckpointStat('checkpoint_handle_duration', 'transaction checkpoint most recent duration for gathering all handles (usecs)', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_handle_duration_apply', 'transaction checkpoint most recent duration for gathering applied handles (usecs)', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_handle_duration_skip', 'transaction checkpoint most recent duration for gathering skipped handles (usecs)', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_handle_skipped', 'transaction checkpoint most recent handles skipped'),
+    CheckpointStat('checkpoint_handle_walked', 'transaction checkpoint most recent handles walked'),
+    CheckpointStat('checkpoint_hs_pages_reconciled', 'number of history store pages that checkpoint caused to be reconciled'),
+    CheckpointStat('checkpoint_pages_read_to_cache', 'number of pages that checkpoint read into the cache'),
+    CheckpointStat('checkpoint_pages_reconciled', 'number of pages that checkpoint caused to be reconciled'),
+    CheckpointStat('checkpoint_pages_visited_internal', 'number of internal pages visited by a checkpoint'),
+    CheckpointStat('checkpoint_pages_visited_leaf', 'number of leaf pages visited by a checkpoint'),
+    CheckpointStat('checkpoint_prep_max', 'transaction checkpoint prepare max time (msecs)', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_prep_min', 'transaction checkpoint prepare min time (msecs)', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_prep_recent', 'transaction checkpoint prepare most recent time (msecs)', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_prep_running', 'transaction checkpoint prepare currently running', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_prep_total', 'transaction checkpoint prepare total time (msecs)', 'no_clear,no_scale'),
+    CheckpointStat('checkpoint_presync', 'number of handles visited after checkpoint writes complete'),
```

Comment:
General comment: I believe FTDC and/or t2 sorts the data by the description strings. So if you want these to appear together you should make sure they all start with 'transaction checkpoint' (or even just 'checkpoint'). Otherwise I think the ones starting with "number of..." will be displayed apart from the others which could be confusing if related stats are apart from each other.

Reply:
I have no idea why I didn't see this comment two weeks ago - apologies for the delayed response. This comment caused me to look at bit deeper at it, and you're right, but: the `CheckpointStat` class prepends all of its descriptions with `checkpoint: `. This leads to descriptions in T2 `checkpoint transaction checkpoint total time (msecs)`, so I'd probably go the other way around and *remove* all of the "checkpoint" and "transaction checkpoint" prefixes. Any thoughts on that?

Reply:
Sure. But the leading text in any group of related stats needs consideration in order to make sure the related ones appear together.